openpype/hosts/max/plugins/publish/extract_tycache.py

```diff
@@ -0,0 +1,197 @@
+    Notes:
+        - TyCache only works for TyFlow Pro Plugin.
+
+    Args:
+        self.export_particle(): sets up all job arguments for attributes
+            to be exported in MAXscript
+
+        self.get_operators(): get the export_particle operator
+
+        self.get_files(): get the files with tyFlow naming convention
+            before publishing
+    """
+
+    order = pyblish.api.ExtractorOrder - 0.2
+    label = "Extract TyCache"
+    hosts = ["max"]
+    families = ["tycache"]
+
+    def process(self, instance):
+        # TODO: let user decide the param
+        start = int(instance.context.data.get("frameStart"))
+        end = int(instance.context.data.get("frameEnd"))
+        self.log.info("Extracting Tycache...")
+
+        stagingdir = self.staging_dir(instance)
+        filename = "{name}.tyc".format(**instance.data)
+        path = os.path.join(stagingdir, filename)
+        filenames = self.get_files(instance, start, end)
+        additional_attributes = instance.data.get("tyc_attrs", {})
+
+        with maintained_selection():
+            job_args = None
+            has_tyc_spline = (
+                True
+                if instance.data["tycache_type"] == "tycachespline"
+                else False
+            )
+            if instance.data["tycache_type"] == "tycache":
+                job_args = self.export_particle(
+                    instance.data["members"],
+                    start, end, path,
+                    additional_attributes,
+                    tycache_spline_enabled=has_tyc_spline)
+            for job in job_args:
+                rt.Execute(job)
+        representations = instance.data.setdefault("representations", [])
+        representation = {
+            'name': 'tyc',
+            'ext': 'tyc',
+            'files': filenames if len(filenames) > 1 else filenames[0],
+            "stagingDir": stagingdir,
+        }
+        representations.append(representation)
+        self.log.info(f"Extracted instance '{instance.name}' to: {filenames}")
+
+        # Get the tyMesh filename for extraction
+        mesh_filename = "{}__tyMesh.tyc".format(instance.name)
+        mesh_repres = {
+            'name': 'tyMesh',
+            'ext': 'tyc',
+            'files': mesh_filename,
+            "stagingDir": stagingdir,
+            "outputName": '__tyMesh'
+        }
+        representations.append(mesh_repres)
+        self.log.info(
+            f"Extracted instance '{instance.name}' to: {mesh_filename}")
+
+    def get_files(self, instance, start_frame, end_frame):
+        """Get file names for tyFlow in tyCache format.
+
+        Set the filenames accordingly to the tyCache file
+        naming extension(.tyc) for the publishing purpose
+
+        Actual File Output from tyFlow in tyCache format:
+        <InstanceName>__tyPart_<frame>.tyc
+
+        e.g. tycacheMain__tyPart_00000.tyc
+
+        Args:
+            instance (str): instance.
```

Comment:
Instance isn't `str` - my linter complains a lot :)

```suggestion
            instance (pyblish.api.Instance): instance.
```